lighthouse-core/gather/connections/connection.js

```diff
@@ -26,6 +26,27 @@ class Connection {
     /** @type {!Map<number, {resolve: function(*), reject: function(*), method: string}>}*/
     this._callbacks = new Map();
     this._eventEmitter = new EventEmitter();
+    this._messageLog = [];
+    this._recordMessages = false;
+  }
+
+  /**
+   * @return {!Array<Object>}
+   */
+  get messageLog() {
+    return this._messageLog;
+  }
+
+  resetMessageLog() {
+    this._messageLog = [];
+  }
+
+  beginRecording() {
```

Comment:
i think we should have a whitelist filter to only record given domains.  that way we could only do Page/Network which is all that WPT needs (and same for Lighthouse in its artifact-driven run).

honestly the API doesn't really need to be configurable.. maybe just set the filter as a property on this class?